tools/taskcluster-worker-runner/protocol/protocol.go

```diff
@@ -22,10 +28,11 @@ type Protocol struct {
 
 func NewProtocol(transport Transport) *Protocol {
 	return &Protocol{
-		transport:    transport,
-		Capabilities: EmptyCapabilities(),
-		callbacks:    make(map[string][]MessageCallback),
-		initialized:  false,
+		transport:          transport,
+		Capabilities:       FullCapabilities(),
```

Comment:
I think some of the misunderstanding in the last review here was caused by this default to all capabilities (the default used to be in the `welcome` callback, below, but was defaulting to FullCapabilities).  As it stands, `Capabilities` will always contain `"shutdown"` because of this default, regardless of the `Add` calls in the providers.

Would it make more sense to default to *no* capabilities and then require other code to call `Add` for the relevant capabilities before starting the protocol?  Or maybe requiring that the capabilities be passed to `NewProtocol`?

Reply:
I don't understand, doesn't `FullCapabilities` return the capabilities listed in `knownCapabilities`? How could `shutdown` always be added?

Reply:
Both of those names are intended to mean "all of the capabilities that exist".  My mistake was thinking that any application using this library would naturally support all of those capabilities -- that's definitely not the case, as here where *sometimes* tc-worker-runner doesn't support `"shutdown"`.

So I'm thinking that the simplest to understand would be to default to no capabilities and require the user of the library to "Add" every capability they do support to the protocol, before starting it.

Reply:
Ok, so can I get rid of `knownCapabilities`?

Reply:
Yeah.  I think I meant to add a check in Add that the capability was in knownCapabilities, but I forgot :)